packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -319,26 +423,84 @@ export class ContextsManager {
     });
   }
 
-  private timeoutId: NodeJS.Timeout | undefined;
-  private dispatchContextsState() {
-    // Debounce: send only the latest value if several values are sent in a short period
-    clearTimeout(this.timeoutId);
-    this.timeoutId = setTimeout(() => {
-      this.apiSender.send(`kubernetes-contexts-state-update`, this.states.getPublished());
-    }, connectTimeout);
+  private setStateAndDispatch(
+    name: string,
+    options: ResourcesDispatchOptions,
+    update: (previous: ContextState) => void,
+  ) {
+    this.states.safeSetState(name, update);
+    this.dispatch({
+      contextsGeneralState: true,
+      currentContextGeneralState: true,
+      resources: options,
+    });
   }
 
-  public getContextsState(): Map<string, ContextState> {
-    return this.states.getPublished();
+  private dispatch(options: DispatchOptions) {
+    if (options.contextsGeneralState) {
+      this.dispatchContextsGeneralState();
+    }
+    if (options.currentContextGeneralState) {
+      this.dispatchCurrentContextGeneralState();
+    }
+    if (options.resources?.pods) {
+      this.dispatchCurrentContextResource('pods');
+    }
+    if (options.resources?.deployments) {
+      this.dispatchCurrentContextResource('deployments');
+    }
   }
 
-  private setStateAndDispatch(name: string, update: (previous: ContextState) => void) {
-    this.states.safeSetState(name, update);
-    this.dispatchContextsState();
+  private generalStateTimeoutId: NodeJS.Timeout | undefined;
+  private dispatchContextsGeneralState() {
+    // Debounce: send only the latest value if several values are sent in a short period
+    clearTimeout(this.generalStateTimeoutId);
+    this.generalStateTimeoutId = setTimeout(() => {
+      this.apiSender.send(`kubernetes-contexts-general-state-update`, this.states.getContextsGeneralState());
+    }, connectTimeout);
+  }
+
+  public getContextsGeneralState(): Map<string, ContextGeneralState> {
+    return this.states.getContextsGeneralState();
   }
 
   private resetConnectionAttempts<T extends KubernetesObject>(options: CreateInformerOptions<T>) {
     options.backoff.reset();
     clearTimeout(options.timer);
   }
+
+  private currentStateTimeoutId: NodeJS.Timeout | undefined;
+  private dispatchCurrentContextGeneralState() {
```

Comment:
missing return type

Reply:
is there a reason to have all methods being private and not protected for example ?

Reply:
I have the habit to declare as private by default. protected would be the default for you?

Reply:
ok good point. The tests I have written for this class are testing it as a black box, only calling the entrypoint `update`, but I get the point